### PR TITLE
fix(gatsby): Only set auto optin flags if not in config

### PR DESCRIPTION
**Pull request by pieh**
## Description

This doesn't change behaviour - this is only to address message that is displayed.

Currently when user explicitly opt-in in config AND site get's picked for auto opt-in we do display message that site was auto opted-in - this is confusing because user already did opt-in manually and we should prefer that over auto opt-in message.

### CODE REVIEW

File: packages/gatsby/src/utils/handle-flags.ts [hunk 1]
```diff
@@ -70,7 +70,9 @@ const handleFlags = (
   availableFlags.forEach(flag => {
     const fitness = flag.testFitness(flag)
 
-    if (configFlags[flag.name] !== false && fitness === `OPT_IN`) {
+    // if user didn't explicitly set a flag (either true or false)
+    // and it qualifies for auto opt-in - add it to optedInFlags
+    if (typeof configFlags[flag.name] === `undefined` && fitness === `OPT_IN`) {
```
Comment (pieh) [inline packages/gatsby/src/utils/handle-flags.ts:75]: actual fix

File: packages/gatsby/src/utils/handle-flags.ts [hunk 2]
```diff
@@ -143,7 +145,7 @@ const handleFlags = (
     if (optedInFlags.size > 0) {
       message += `\n\n`
       message += `We're shipping new features! For final testing, we're rolling them out first to a small % of Gatsby users
-and your site was automatically choosen as one of them. With your help, we'll then release them to everyone in the next minor release.
+and your site was automatically chosen as one of them. With your help, we'll then release them to everyone in the next minor release.
```
Comment (pieh) [inline packages/gatsby/src/utils/handle-flags.ts:148]: randomly spoted typo

Comment (sidharthachatterjee) [reply packages/gatsby/src/utils/handle-flags.ts:148]: You mean randomly _spotted_ typo :trollface:

**Review by sidharthachatterjee — APPROVED** [review]
Nice work! Thanks 🥧

File: packages/gatsby/src/utils/handle-flags.ts (hunk 2 above)
Comment (pieh) [reply packages/gatsby/src/utils/handle-flags.ts:148]: hahaha, oh the irony xD

**Pull request merged**